addons/google_calendar/models/google_sync.py

```diff
@@ -149,6 +149,7 @@ def _sync_google2odoo(self, google_events: GoogleEvent, default_reminders=()):
         """
         existing = google_events.exists(self.env)
         new = google_events - existing - google_events.cancelled()
+        write_dates = self._context.get('write_dates', {})
```

Comment:
Actually I think it is possible to achieve the same result without playing with the context (usually a bad practice because hard to debug, could be erase or not propagated etc).

```suggestion
        odoo_records = google_events.odoo_ids(self.env)
        write_dates = {r.id: r.write_date for r in odoo_records}
```

What do you think ?